src/linker/Linker/Annotations.cs

```diff
@@ -678,7 +678,10 @@ public void EnqueueVirtualMethod (MethodDefinition method)
 			if (!method.IsVirtual)
 				return;
 
-			if (FlowAnnotations.RequiresVirtualMethodDataFlowAnalysis (method) || HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (method))
+			// Implementations of static interface methods are not virtual and won't show up here.
+			// If the base static interface methods doesn't have annotations, the implementation still might and we need to warn.
+			// To account for this, we'll make sure to check all static interface method's overrides
+			if (FlowAnnotations.RequiresVirtualMethodDataFlowAnalysis (method) || HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (method) || method.IsStatic)
```

Comment:
This should only be called on the static interface methods that are virtual, but will never get called on their implementations since they're not virtual (it won't get called on them once I remove the IsStatic check in MarkStep.ProcessMethod). To account for this, I had it check all implementations of all static interface methods.

However, instead of always checking all static interface methods in ValidateVirtualMethodAnnotationsStep, we could instead check all the overrides for static virtual methods here and only add the relevant methods to ValidateVirtualMethodAnnotationsStep. That would probably be faster, so I'll do that instead.